.github/workflows/ci.yml

```diff
@@ -303,7 +303,52 @@ jobs:
             --env IREE_VULKAN_F16_DISABLE=0 \
             --env IREE_CUDA_DISABLE=0 \
             --env IREE_NVIDIA_GPU_TESTS_DISABLE=0 \
+            --env IREE_NVIDIA_SM80_TESTS_DISABLE=1 \
             --env CTEST_PARALLEL_LEVEL=2 \
+            --env IREE_CTEST_LABEL_REGEX='^requires-gpu|^driver=vulkan$|^driver=cuda$' \
+            --gpus all \
+            --env NVIDIA_DRIVER_CAPABILITIES=all \
+            gcr.io/iree-oss/nvidia@sha256:1717431fd46b8b1e96d95fa72508e3e3eacb5c95f1245b9b7dbeec23ae823d02 \
+            bash -euo pipefail -c \
+              "./build_tools/scripts/check_cuda.sh
+              ./build_tools/scripts/check_vulkan.sh
+              ./build_tools/cmake/ctest_all.sh ${BUILD_DIR}"
+
+  test_nvidia_sm80:
```

Comment:
I think in general the benefit is just reducing boilerplate code. But it's probably not much clearer.

Reply:
Let me try it out and see how it looks. I'll push a commit in a moment

Reply:
While we're waiting to see if the matrix is correct, feel free to comment on whether you think it's clearer or you'd do it differently. I did a bit of a weird thing with generating files rather than having a ton of matrix parameters.

Reply:
I'm on the fence... the premise of a job matrix is nice, but there's still quite a bit of boilerplate for a fairly straightforward branch between the configurations. I think there's also value in having the "basic" jobs not contain any logic (easier to read to see what they're doing, easier to fork).

I think at least the `- runner-type: gpu` and `- runner-type: a100` parts of the matrix are an improvement... might be better to keep cpu out of the matrix though?

Reply:
+1 on not having gpu and cpu in the same matrix to avoid branching

Reply:
Ok I gave just GPU matrix a try. PTAL. Also GitHub appears to dislike heredocs, so that's annoying